.eslintrc.cjs

```diff
@@ -1,23 +1,55 @@
 module.exports = {
+  extends: [
+    'plugin:@typescript-eslint/recommended-type-checked',
+    'plugin:@typescript-eslint/stylistic-type-checked',
+    'prettier',
+  ],
   parser: '@typescript-eslint/parser',
-  extends: ['plugin:@typescript-eslint/recommended', 'prettier'],
+  parserOptions: {
+    project: ['./packages/*/tsconfig.json', './tsconfig.eslint.json'],
+    tsconfigRootDir: __dirname,
+  },
   plugins: ['@typescript-eslint', 'prettier', 'no-only-tests'],
   rules: {
+    // These off/configured-differently-by-default rules fit well for us
+    '@typescript-eslint/no-unused-vars': ['error', { ignoreRestSiblings: true }],
+    'no-only-tests/no-only-tests': 'error',
+    '@typescript-eslint/no-shadow': ['error'],
+    'no-console': 'warn',
+
+    /*
+    interface: 404 results in 198 files
+    type: 129 results in 73 files
+    */
+    '@typescript-eslint/consistent-type-definitions': 'off',
```

Comment:
It looks like you've _somewhat_ solidified on preferring `interface` over `type`, but both have a good amount of usage. Is using one over the other when possible something you'd want to standardize?

https://typescript-eslint.io/rules/consistent-type-definitions

Reply:
I usually prefer `interface` over `type`. I think some of us prefer the other way around too. Maybe we can enforce this in a later PR, or if we decide on a style here.

Reply:
Usually, I go the other way around. 

`type` by default, and `interface` if I need to use its features:
- lazy loading
- `extends`


I don't have strong opinions, though.

Reply:
I'm with Ema on that too. `type` is more ergonomic to write in my opinion, and I like that it avoids interface merging. Loose opinion strongly held :)

Reply:
I don't have a strong reason picking `interface` over `type`, other than I use them the same way as `function foo() {}` vs `const foo = () => {}`. E.g. `interface` for typing object shapes, `type` for quick one-offs. But I also somewhat like that `interface` generally provides nicer TS intellisense message, especially when unions/extends are used.

Reply:
I think we should avoid doing an upgrade and changing (or adding) lint rules in the same PR. Those are two separate ideas that should be discussed separately.

Reply:
I'll leave this as a todo then - agreed that if there isn't team consensus, this is not a great place to come to one. Thanks!

@bholmesdev looking forward to a whiteboard video on the nuanced differences between the two 😉

Reply:
Note that changing from `interface` to `type` can be a breaking change, both due to inability to merge declarations and potential performance regressions.

Reply:
FWIW at this point `type` has been optimized pretty significantly. I can't remember the last time I've seen it be a performance concern in any code (and would love to be pointed to a demonstrable example!). The TS wiki page is out of date.